graphql-kotlin-spring-server/src/test/kotlin/com/expediagroup/graphql/spring/context/GraphQLContextFactoryIT.kt

```diff
@@ -0,0 +1,93 @@
+import com.expediagroup.graphql.spring.execution.GRAPHQL_CONTEXT_FILTER_ODER
+import com.expediagroup.graphql.spring.execution.GraphQLContextFactory
+import com.expediagroup.graphql.spring.model.GraphQLRequest
+import com.expediagroup.graphql.spring.operations.Query
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.reactor.ReactorContext
+import org.junit.jupiter.api.Test
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.boot.autoconfigure.EnableAutoConfiguration
+import org.springframework.boot.test.context.SpringBootTest
+import org.springframework.context.annotation.Bean
+import org.springframework.context.annotation.Configuration
+import org.springframework.core.annotation.Order
+import org.springframework.http.MediaType
+import org.springframework.http.server.reactive.ServerHttpRequest
+import org.springframework.http.server.reactive.ServerHttpResponse
+import org.springframework.test.web.reactive.server.WebTestClient
+import org.springframework.web.server.WebFilter
+import kotlin.coroutines.coroutineContext
+
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT, properties = ["graphql.packages=com.expediagroup.graphql.spring.context"])
+@EnableAutoConfiguration
+class GraphQLContextFactoryIT(@Autowired private val testClient: WebTestClient) {
+
+    @Test
+    fun `verify context is generated and available to the GraphQL execution`() {
+        testClient.post()
+            .uri("/graphql")
+            .header("X-First-Header", "JUNIT_FIRST")
+            .header("X-Second-Header", "JUNIT_SECOND")
+            .accept(MediaType.APPLICATION_JSON)
+            .contentType(MediaType.APPLICATION_JSON)
+            .bodyValue(GraphQLRequest("query { context { first second } }"))
+            .exchange()
+            .expectBody()
+            .jsonPath("$.data.context").exists()
+            .jsonPath("$.data.context.first").isEqualTo("JUNIT_FIRST")
+            .jsonPath("$.data.context.second").isEqualTo("JUNIT_SECOND")
+            .jsonPath("$.errors").doesNotExist()
+            .jsonPath("$.extensions").doesNotExist()
+    }
+
+    @Configuration
+    class GraphQLContextFactoryConfiguration {
+
+        @Bean
+        fun query(): Query = ContextualQuery()
+
+        @Bean
+        @ExperimentalCoroutinesApi
+        fun customContextFactory(): GraphQLContextFactory<CustomContext> = object : GraphQLContextFactory<CustomContext> {
+            override suspend fun generateContext(request: ServerHttpRequest, response: ServerHttpResponse): CustomContext {
+                val firstValue = coroutineContext[ReactorContext]?.context?.get<String>("firstFilterValue")
+                return CustomContext(
+                    first = firstValue,
+                    second = request.headers.getFirst("X-Second-Header") ?: "DEFAULT_SECOND"
+                )
+            }
+        }
+
+        @Bean
+        @Order(GRAPHQL_CONTEXT_FILTER_ODER - 10)
```

Comment:
where does this `10` come from?

Reply:
`10` doesn't matter, all what we care about is that final order value is < `GRAPHQL_CONTEXT_FILTER_ORDER`

Reply:
can we just do a `-1` then. Justifies the intent much clearly. `10` gives me an impression of a magic number if I am looking at the code out of the blue.